src/main/java/components/Parser.java

```diff
@@ -0,0 +1,108 @@
+package components;
+
+import commands.*;
+
+import java.text.ParsePosition;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.InputMismatchException;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+public class Parser {
+    private static SimpleDateFormat dateFormatter = new SimpleDateFormat("dd/MM/yyyy HHmm");
+
+    /**
+     * Returns a date from a string.
+     */
+    private static Date parseAsDate(String str) {
+        return dateFormatter.parse(str, new ParsePosition(0));
+    }
+
+    public static Command parse(String fullCommand) {
+        Scanner sc = new Scanner(fullCommand);
+        String firstWord = sc.next();
+        switch (firstWord) {
+            case "find":
+                try {
+                    String keyword = sc.nextLine();
```

Comment:
You could choose to abstract away the parsing of the command arguments to prevent class bloat